### DISCUSSION TITLE
Integration with SvelteKit Preprocessor

### DISCUSSION THREAD

**Discussion by vhscom** [ivanhofer/typesafe-i18n#155] [General]
Thanks for your hard work on this project, Ivan. The thought and consideration you put in is clear from the comprehensive README and your [Svelte Summit talk](https://www.youtube.com/watch?v=9_LoEY8hO44) a few hours ago. I found this project after clicking through from your helpful comments in https://github.com/sveltejs/kit/issues/553 and wanted to know if you have any plans of integrating it into the Svelte preprocessing pipeline similar to https://github.com/pngwn/mdsvex.

It seems that may be the case based on https://github.com/ivanhofer/typesafe-i18n/discussions/89 but I wasn't sure if you had something more general in mind given this project is targeted at a number of different use cases (e.g. Vue, React, etc.). Knowing this will help me determine if I should invest more time with this project or look at more Svelte-specific solutions such as https://svelte-intl-precompile.com/.

Thanks again for your great work!

**Comment by ivanhofer** [comment]
Hi @vhscom thanks for opening this discussion.
I really want to improve the library and optimize the generated output. You already have referenced the correct discussion #89.
This would probably mean to come up with a general solution that parses the AST and optimizes the translation files. This solution then can be used inside a rollup or vite plugin and probably also in a Svelte preprocessor. Ideally this would work with all frameworks.

Unfortunately I currently don't have the available ressources to dive into this topic. Maybe someone could help me out or it has to wait for the secont half of this year.
Since `typesafe-i18n` already produces a small footprint, I (personally) don't see this as a critical must-have feature.

If the preprocessing thing is something you really need and you can't wait, then you probably should use `svelte-intl-precompile`. You have to choose between saving a few bytes in the bundle or a having a great developer-experience (personal opinion). `typesafe-i18n` offers something no other i18n solution currently provides: it's strong typesafety.
